### PR TITLE
Implemented the RSS feeds endpoint

### PR DESCRIPTION
**Pull request by devos50**
And another one! This one implements the endpoint to fetch RSS feeds used by your channel.

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis ready for review

**Comment by lfdversluis** [comment]
Looks ok to me

**Comment by lfdversluis** [comment]
One method not covered in my_channel_endpoint.py ?

**Comment by devos50** [comment]
@lfdversluis I've changed something in a uncovered method and got the coverage diff complaining so I've covered it now. That uncovered method shouldn't have been merged in the first place.

**Comment by devos50** [comment]
@whirm ready for review

**Comment by whirm** [comment]
Have a look at the pylint report, some of that stuff should be fixed.

**Comment by devos50** [comment]
@whirm Ah didn't notice them. I fixed the most important issues.

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 1]
```diff
@@ -29,9 +31,10 @@ class MyChannelEndpoint(MyChannelBaseEndpoint):
     """
 
     def getChild(self, path, request):
-        child_handler_dict = {"overview": MyChannelOverviewEndpoint, "torrents": MyChannelTorrentsEndpoint}
+        child_handler_dict = {"overview": MyChannelOverviewEndpoint, "torrents": MyChannelTorrentsEndpoint,
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: Why aren't you using `putChild()` instead? https://twistedmatrix.com/trac/browser/tags/releases/twisted-14.0.0/twisted/web/resource.py#L209

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 2]
```diff
@@ -90,3 +93,30 @@ def render_GET(self, request):
...
+        my_channel_id = self.channel_db_handler.getMyChannelId()
+        channel_obj = self.session.lm.channel_manager.get_my_channel(my_channel_id)
+        if my_channel_id is None or channel_obj is None:
+            return MyChannelBaseEndpoint.return_404(request)
+
+        rss_list = channel_obj.get_rss_feed_url_list()
+        request.setHeader('Content-Type', 'text/json')
+        feeds_list = []
+        for rss_item in rss_list:
+            feeds_list.append({'url': rss_item})
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: You could use a list comprehension instead of these 3 lines of code.

**Comment by whirm** [comment]
BTW, I don't know the details, but @LipuFei refactored the channel managing system so you can have several channels instead of one. I think the API (and the new GUI) should reflect that.

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py (hunk 2 above)
Comment (LipuFei) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: @devos50 @whirm I think this is right place. The non-GUI channel and RSS code are now in `Tribler/Core/Modules/channel`.

**Comment by devos50** [comment]
@whirm implemented your feedback. I will change the `getChild` occurrences in other files in upcoming PRs.

I'm aware that there it is possible to have several channels, however, I see this as an enhancement. I propose to discuss multiple channel management with @synctext and keep it simple for now.

**Comment by whirm** [comment]
I'm fine with discussing how the GUI should look, but I thing the API should reflect the fact that more than one channel can be managed before we release the first stable version of the API.

**Comment by whirm** [comment]
tunnel community tests getting stuck again

**Comment by whirm** [comment]
retest this please

**Comment by devos50** [comment]
@whirm you want me to do this in this PR?

**Comment by whirm** [comment]
No, because we already have code that exposes a single channel API and that would need to be changed too, It just needs to happen before we've got stuff using the current API to avoid having to refactor that too.

**Comment by devos50** [comment]
@whirm ok I will update my GUI PR to add a checkbox for this ;)

**Comment by whirm** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Tunnel tests timing out again...

**Comment by lfdversluis** [comment]
succeeded now

**Comment by devos50** [comment]
@whirm the builders are up again

**Comment by whirm** [comment]
Looks good, squash!

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm done

**Pull request merged**